[charlenelau-20] iP

charlenelau-20 opened the pull request:
### ChitChat

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

ChitChat frees your mind of having to remember things you need to do.

It's:

- text-based
- easy to learn
- ~~fast~~ _SUPER_ fast to use

All you need to do is:

1. download it from [here](https://github.com/charlenelau-20/ip/releases/tag/v0.1).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :white_check_mark:

and it's...**FREE** :bangbang:

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too!
Here's the `main` method:

```java
public static void main(String[] args) {
    new ChitChat("data/chitchat.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

charlenelau-20 committed:
add ChitChat.java

charlenelau-20 committed:
remove Duke.java

charlenelau-20 committed:
update ChitChat.java

charlenelau-20 committed:
add echoing feature (Level-1) to ChitChat.java

charlenelau-20 committed:
(Level-2) update ChitChat.java to read user inputs and display as list

charlenelau-20 committed:
add Task class (Task.java)
update ChitChat.java with Mark as Done/Not Done feature for Level-3

charlenelau-20 committed:
todos, events, deadlines features added

charlenelau-20 committed:
automated text UI testing (Level-0 to Level-4)

charlenelau-20 committed:
implement error handling

charlenelau-20 committed:
add delete feature

charlenelau-20 committed:
update test files, minor fix of error message

charlenelau-20 committed:
Add Storage.java

Task.java: Modify to accomodate Storage.java

Todo.java: Modify to accomodate Storage.java

Deadline.java: Modify to accomodate Storage.java

Event.java: Modify to accomodate Storage.java

ChitChat.java: Update to save tasks

charlenelau-20 committed:
Update files to include header comments

charlenelau-20 committed:
ChitChat.java: Update with date and time logic
Deadline.java: Update with date and time logic
Event.java: Update with date and time logic
Storage.java: Update with date and time logic

charlenelau-20 committed:
ChitChat.java: update line length and make code more readable
Deadline.java: handle exceptions for date and time formatting
Event.java: handle exceptions for date and time formatting
EXPECTED.TXT: update expected output
input.txt: update sample inputs

charlenelau-20 committed:
Add Ui.java
Add TaskList.java
Add Parser.java
ChitChatException.java: Add header comments
ChitChat.java: Update for OOP
Storage.java: Update for OOP
bug fix: Remove blank line spaces

charlenelau-20 committed:
Move files to respective packages

charlenelau-20 committed:
Merge branch 'branch-A-Packages'

charlenelau-20 committed:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

charlenelau-20 committed:
build.gradle: Add to plugins and dependencies
bug fix: Correct welcome message

charlenelau-20 committed:
Add checkstyle.xml
Add suppressions.xml
style: update code so that it aligns with the code style rules

charlenelau-20 committed:
Add ChitChatTest.java
Add ParserTest.java
Add ChitChatExceptionTest.java
Add StorageTest.java
Add DeadlineTest.java
Add EventTest.java
Add TaskListTest.java
Add TaskTest.java
Add TodoTest.java

charlenelau-20 committed:
build.gradle: update to make the app an executable JAR file

zhenglong1603 commented on gradle/wrapper/gradle-wrapper.properties:
consider adding the gradle file to keep the repo cleaner and for devops purposes!

zhenglong1603 commented on src/main/java/chitchat/task/Task.java:
```diff
@@ -0,0 +1,71 @@
+ * Each task consists of a description and a completion status.
+ * Tasks can be marked as done or not done.
+ */
+public class Task {
```
consider making this an abstract class! you've already got the three task types, there's probably no reason for a bare Task object to be instantiated at this point.

awlarpi replied:
Yes I agree, abstract class will be a better fit.

zhenglong1603 commented on src/main/java/chitchat/command/Parser.java:
```diff
@@ -0,0 +1,160 @@
+     */
+    public void parseCommand(String input) {
+        try {
+            if (input.equals("list")) {
```
maybe consider switch and case instead of if else statements?

Shanyey replied:
agree with zhenglong, switch case could make code more readable and shorter

zhenglong1603 commented on src/main/java/chitchat/command/Parser.java:
```diff
@@ -0,0 +1,160 @@
+                taskList.markTask(taskIndex);
+
+                ui.showLine();
+                System.out.println("Nice! I've marked this task as done:\n  " + taskList.getTasks().get(taskIndex));
```
you can consider abstracting all these responses and printing to the UI classs. maybe pass the message and the taskIndex to UI and have a method in UI to build the showing of lines and whole message :)

awlarpi replied:
I think the function is too long, you can consider abstracting but perhaps creating a Command class with types of commands that can be executed.

zhenglong1603 reviewed (commented):
Overall good effort! Hope to see the tests being built and keep up the gd work!:)

charlenelau-20 committed:
Parser.java: Add comments

charlenelau-20 committed:
style: Edit code layout

charlenelau-20 committed:
Parser.java: Update to handle find command
TaskList.java: Add findTasks method
style: edit comments

charlenelau-20 committed:
Merge branch 'branch-A-JavaDoc'

charlenelau-20 committed:
Merge branch 'branch-A-CodingStandard'

charlenelau-20 committed:
Merge branch 'branch-Level-9'

charlenelau-20 committed:
style: improve code formatting

Shanyey commented on src/main/java/chitchat/command/Parser.java:
```diff
@@ -0,0 +1,193 @@
+package chitchat.command;
```
Would renaming this package to chitchat.parser make it easier to read? You may consider adding a command class in the future to handle the comments as well

Shanyey commented on src/main/java/chitchat/command/Parser.java:
```diff
@@ -0,0 +1,193 @@
+                taskList.unmarkTask(taskIndex);
+
+                ui.showLine();
+                System.out.println("OK, I've marked this task as not done yet:\n  "
```
Can consider shifting the print statements into the ui class to keep the parser class simplier

Shanyey commented on src/main/java/chitchat/storage/Storage.java:
```diff
@@ -0,0 +1,115 @@
+                LocalDateTime by = LocalDateTime.parse(parts[3], DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+                return new Deadline(description, by, isDone);
+            case "E":
+                LocalDateTime from = LocalDateTime.parse(parts[3], DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
```
Can consider declaring the datetime formatter once and passing it in to the parser

Shanyey commented on src/main/java/chitchat/task/Event.java:
```diff
@@ -0,0 +1,61 @@
+    @Override
+    public String toFileFormat() {
+        return "E | " + super.toFileFormat() + " | " + from.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"))
+                + " | " + to.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
```
Perhaps you could declare the datetimeformatter and reuse for the 4 areas as well

Shanyey commented on src/main/java/chitchat/task/Task.java:
```diff
@@ -0,0 +1,71 @@
+     *
+     * @return A string representing the task in file format.
+     */
+    public String toFileFormat() {
```
If you were to change task to abstract class as suggested by zhenglong, you can consider making this method an abstract method if each task has a different format of saving in file format

Shanyey reviewed (commented):
Good coding standard and use of javadocs
parser seems to be quite long though, can consider extracting out the commands.
LGTM

awlarpi commented on src/main/java/chitchat/task/TaskList.java:
```diff
@@ -0,0 +1,154 @@
+     * @throws ChitChatException If the index is out of bounds.
+     */
+    public void deleteTask(int index) throws ChitChatException {
+        if (index < 0 || index >= tasks.size()) {
```
Nice error handling!

awlarpi commented on src/test/java/chitchat/storage/StorageTest.java:
```diff
@@ -0,0 +1,61 @@
+    private TaskList taskList;
+
+    @BeforeEach
+    void setUp() {
```
Nice way to run setup code.

awlarpi reviewed (commented):
Looks good to me.

awlarpi reviewed (commented):
Looks good, I see no serious naming violations.

charlenelau-20 committed:
Add GUI to ChitChat

charlenelau-20 committed:
Merge branch 'branch-Level-10'

charlenelau-20 committed:
Generate new jar file with JavaFX libraries

charlenelau-20 committed:
Add assertions

charlenelau-20 committed:
Delete unused code
bug fix: fix to load saved task status correctly

charlenelau-20 committed:
Add comprehensive assertions to Storage.java

The initial assertions added to the Storage class were incomplete
and lacked proper input validation.

Added assertions for null checks as well as empty strings.

Making the assertions more comprehensive helps to catch programming
errors and bugs earlier.

charlenelau-20 committed:
Merge pull request #1 from charlenelau-20/branch-A-Assertions

Add assertions

charlenelau-20 committed:
Merge branch 'master' into branch-A-CodeQuality

charlenelau-20 committed:
Merge pull request #2 from charlenelau-20/branch-A-CodeQuality

Check code quality

charlenelau-20 committed:
Add help command

A help command allows users to view the list of commands, how to use
them and what each command does.

Introducing a help command enhances user-friendliness by providing in-app
guidance for users.

charlenelau-20 committed:
Merge branch 'branch-C-Help'

charlenelau-20 committed:
Add improvements to GUI

- Make icons circular
- Make error message color different from normal responses
- Make chatbot message color different from user

These improvements to the GUI enhance user experience.

charlenelau-20 committed:
Merge branch 'branch-A-BetterGui'

charlenelau-20 committed:
Remove commented code

charlenelau-20 committed:
Parser.java: Modify to simplify parseCommand method

MainWindow.java: Remove unused code

charlenelau-20 committed:
Main.java: Edit to have app name appear in title bar of GUI

charlenelau-20 committed:
Edit handleHelp() method

charlenelau-20 committed:
Update test files with some test cases

charlenelau-20 committed:
Add Ui.png

charlenelau-20 committed:
Update user guide

charlenelau-20 committed:
Launcher.java: Edit header comment

Parser.java: Add header comments to the private methods

charlenelau-20 committed:
README.md: Add credits for icon images

charlenelau-20 committed:
TaskList.java: Remove unused code

charlenelau-20 committed:
README.md: Minor fix

charlenelau-20 committed:
Add citation comments

charlenelau-20 committed:
Add citation comment
